code/game/objects/structures/inflatable.dm

```diff
@@ -0,0 +1,298 @@
+	set name = "Deflate"
+	set category = "Object"
+	set src in oview(1)
+
+	if(isobserver(usr) || usr.restrained() || !usr.Adjacent(src))
+		return
+
+	verbs -= /obj/structure/inflatable/verb/hand_deflate
+	deflate()
+
+/obj/structure/inflatable/proc/attack_generic(var/mob/user, var/damage, var/attack_verb)
+	health -= damage
+	if(health <= 0)
+		spawn(1) deflate(1)
+	return 1
+
+/obj/structure/inflatable/door
+	name = "inflatable door"
+	density = 1
+	anchored = 1
+	opacity = 0
+
+	icon_state = "door_closed"
+	undeploy_path = /obj/item/inflatable/door
+
+	var/state = 0 //closed, 1 == open
+	var/isSwitchingStates = 0
+
+/obj/structure/inflatable/door/attack_ai(mob/user as mob) //those aren't machinery, they're just big fucking slabs of a mineral
+	if(isAI(user)) //so the AI can't open it
+		return
+	else if(isrobot(user)) //but cyborgs can
+		if(get_dist(user,src) <= 1) //not remotely though
+			return TryToSwitchState(user)
+
+/obj/structure/inflatable/door/attack_hand(mob/user as mob)
+	return TryToSwitchState(user)
+
+/obj/structure/inflatable/door/Cross(atom/movable/mover, turf/target, height=0, air_group=0)
+	if(air_group)
+		return state
+	if(istype(mover, /obj/effect/beam))
+		return !opacity
+	return !density
+
+/obj/structure/inflatable/door/proc/TryToSwitchState(atom/user)
+	if(isSwitchingStates) return
+	if(ismob(user))
+		var/mob/M = user
+		if(M.client)
+			if(iscarbon(M))
+				var/mob/living/carbon/C = M
+				if(!C.handcuffed)
+					SwitchState()
+			else
+				SwitchState()
+	else if(istype(user, /obj/mecha))
+		SwitchState()
+
+/obj/structure/inflatable/door/proc/SwitchState()
+	if(state)
+		Close()
+	else
+		Open()
+	update_nearby_tiles()
+
+/obj/structure/inflatable/door/proc/Open()
+	isSwitchingStates = 1
+	flick("door_opening",src)
+	playsound(loc, 'sound/effects/attackblob.ogg', 75, 1)
+	sleep(10)
+	density = 0
+	opacity = 0
+	state = 1
+	update_icon()
+	isSwitchingStates = 0
+
+/obj/structure/inflatable/door/proc/Close()
+	isSwitchingStates = 1
+	flick("door_closing",src)
+	playsound(loc, 'sound/effects/attackblob.ogg', 75, 1)
+	sleep(10)
+	density = 1
+	opacity = 0
+	state = 0
+	update_icon()
+	isSwitchingStates = 0
+
+/obj/structure/inflatable/proc/update_nearby_tiles()
+	if (isnull(air_master))
+		return 0
+
+	var/T = loc
+
+	if (isturf(T))
+		air_master.mark_for_update(T)
+
+	return 1
+
+/obj/structure/inflatable/door/update_icon()
+	if(state)
+		icon_state = "door_open"
+	else
+		icon_state = "door_closed"
+
+/obj/structure/inflatable/door/deflate(var/violent=0)
+	playsound(loc, 'sound/machines/hiss.ogg', 75, 1)
+	if(violent)
+		visible_message("[src] rapidly deflates!")
+		var/obj/item/inflatable/door/torn/R = new /obj/item/inflatable/door/torn(loc)
+		src.transfer_fingerprints_to(R)
+		qdel(src)
+	else
+		visible_message("[src] slowly deflates.")
+		spawn(50)
+			var/obj/item/inflatable/door/R = new /obj/item/inflatable/door(loc)
+			src.transfer_fingerprints_to(R)
+			qdel(src)
+
+/obj/item/inflatable/torn
```

Comment:
Will there be a patch kit like duct tape, or are they just something to be tossed in the trash?